bundle/manifest/manifest.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// Package manifest implements helper functions for the stored manifest.
+package manifest
+
+import (
+	"context"
+	"fmt"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/util"
+)
+
+var bundlesBasePath = storage.MustParsePath("/system/bundles")
+
+// Note: As needed these helpers could be memoized
+
+func bundlePath(name string) storage.Path {
+	return append(bundlesBasePath, name)
+}
+
+func manifestPath(name string) storage.Path {
+	return append(bundlesBasePath, name, "manifest")
+}
+
+func rootsPath(name string) storage.Path {
+	return append(bundlesBasePath, name, "manifest", "roots")
+}
+
+func revisionPath(name string) storage.Path {
+	return append(bundlesBasePath, name, "manifest", "revision")
+}
+
+// ReadBundleNames will return a list of bundle names which have had their metadata stored.
+func ReadBundleNames(ctx context.Context, store storage.Store, txn storage.Transaction) ([]string, error) {
+	value, err := store.Read(ctx, txn, bundlesBasePath)
+	if err != nil {
+		return nil, err
+	}
+
+	bundleMap, ok := value.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("corrupt manifest roots")
+	}
+
+	bundles := make([]string, len(bundleMap))
+	idx := 0
+	for name := range bundleMap {
+		bundles[idx] = name
+		idx++
+	}
+	return bundles, nil
+}
+
+// Write the manifest into the storage. This function is called when
+// the bundle is activated.
+func Write(ctx context.Context, store storage.Store, txn storage.Transaction, name string, m bundle.Manifest) error {
+
+	var value interface{} = m
```

Comment:
[nit] Interesting way of putting it. It's equivalent to `value := interface{}(m)`, right?

Reply:
Yea I think they are pretty much the same